rfcs/text/0005-host-metric-fields.md

```diff
@@ -26,6 +26,17 @@ Proposed 7 new fields are:
 <!--
 Stage 1: Describe at a high level how this change affects fields. Which fieldsets will be impacted? How many fields overall? Are we primarily adding fields, removing fields, or changing existing fields? The goal here is to understand the fundamental technical implications and likely extent of these changes. ~2-5 sentences.
 -->
+This RFC calls for the addition of host fields to collect basic monitoring metrics from a host or VM such as CPU, network and disk.
+
+| field | type | description |
+| --- | --- | --- |
+| `host.cpu.pct` | scaled_float | Percent CPU used. This value is normalized by the number of CPU cores and it ranges from 0 to 1. |
```

Comment:
It's important to specify the `scaling_factor` so that we know how the value should be interpreted.

Reply:
Can we document how this cpu percentage works on a server with multiple CPU cores.
Example, If I have a server with 12 CPU cores, will this "CPU percentage" max to `1200%` or to `100%`?

Reply:
I believe the way that the endpoint team has done this in the past is the 12 CPU cores has a max of `1200.0123...` right @ferullo ?

Reply:
Thanks for the input! This `host.cpu.pct` will be the normalized value ranges from 0 to 1 only. If the server has 12 CPU cores, then this normalized value will be `total cpu percentage`/12.

Reply:
Would that make `scaling_factor: 100`?

This would be the first time a `scaled_float` is defined in ECS, and we'll need to add support for the defining `scaling_factor` in the schema. I don't see any issue with the addition - just noting that dependency.

Reply:
@ebeahan I was thinking to keep `scaleing_factor` as default value. So value 0.12 for example will be stored into ES as 0.12 itself. Maybe I should just use `float` here instead? Thanks!

Reply:
On the `scaling_factor`, it's a required parameter for this datatype, there's no default value here.

Note that the source doc is not affected by it, and would always contain the full float value.

A scaling factor of 100 would mean 0.123 (for 12.3%) would only store 0.12 in the index, for aggregations & so on. If we want to aggregate on full digit percents, a scaling_factor of 100 is appropriate.

@kaiyan-sheng or @cyrille-leclerc is 100 appropriate for this use case? Who can chime in on how observability intends to query this field?

<details>
<summary>Experiment with scaled_float</summary>

I was hoping that `scaled_float` being backed by a long meant it wouldn't have float artifacts, but with the simple test below, the last two documents ingested show up with `...0000001` in the aggregation 🤷‍♂️ 

```
PUT sfloat-1
{ "mappings" : { "properties": {
  "pct" : { "type" : "scaled_float", "scaling_factor": 100 }
}}}
PUT sfloat-1/_doc/1
{ "pct": 0.12 }
PUT sfloat-1/_doc/2
{ "pct": 0.1234 }


PUT sfloat-2
{ "mappings" : { "properties": {
  "pct" : { "type" : "scaled_float", "scaling_factor": 10000 }
}}}
PUT sfloat-2/_doc/1
{ "pct": 0.12 }
PUT sfloat-2/_doc/2
{ "pct": 0.1234 }

GET sfloat-*/_search
GET sfloat-*/_search
{ "size": 0, "aggs" : {
    "percentages" : { "terms" : { "field" : "pct" }
}}}
```

</details>

Reply:
@webmat Thank you for the explanation!! I think I would prefer a scale_factor of 1000 in this case so for CPU usage, we are keeping one decimal when displaying it in a percentage format. For example, I would like to store 0.1234 to be 0.123 in the index so it will show as 12.3% in percentage. But if this is a concern with saving space for storing these values, scale_factor of 100 is also fine for me. @exekias WDYT?

Reply:
We can try to do some testing and compare sizes, 100 has always looked a bit limiting, some more precision would be good